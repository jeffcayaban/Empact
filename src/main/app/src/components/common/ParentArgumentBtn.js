import React from "react";
import { Button } from "react-bootstrap";

/**
 * Displays a button for viewing an argument's parent argument.
 */

class ParentArgumentBtn extends React.PureComponent {

    render() {
        const { parentArgId, showArg } = this.props;

        if (parentArgId !== null && parentArgId !== undefined) {
            // Assumes that the parent argument exists.

            return (
                <Button style={{ marginRight: '10px' }} onClick={() => showArg()}>
                    Parent Argument
                </Button>
            )
        } else {
            // Assumes that the parent argument does not exists.

            return null;
        }
    }
}

export default ParentArgumentBtn;