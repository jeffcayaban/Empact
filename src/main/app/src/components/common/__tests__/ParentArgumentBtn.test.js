import React from 'react';
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import ParentArgumentBtn from "../ParentArgumentBtn";

describe('ParentArgumentBtn', () => {

    it('should render successfully', () => {
        const component = shallow(<ParentArgumentBtn
            parentArgId={"testId"}
            showArg={jest.fn()}
        />);

        expect(toJson(component)).toMatchSnapshot();
    });

    it('should render nothing', () => {
        const component = shallow(<ParentArgumentBtn
            showArg={jest.fn()}
        />);

        expect(toJson(component)).toMatchSnapshot();
    });

});
