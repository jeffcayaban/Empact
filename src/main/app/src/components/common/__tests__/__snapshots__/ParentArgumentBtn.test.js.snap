// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParentArgumentBtn should render nothing 1`] = `""`;

exports[`ParentArgumentBtn should render successfully 1`] = `
<Button
  active={false}
  block={false}
  bsClass="btn"
  bsStyle="default"
  disabled={false}
  onClick={[Function]}
  style={
    Object {
      "marginRight": "10px",
    }
  }
>
  Parent Argument
</Button>
`;
